File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:

        left = 0
        countZeros = 0
        res = 0

        for right in range(len(nums)):

            if nums[right] == 0:
                countZeros += 1

                while countZeros > 1:
                    if nums[left] == 0:
                        countZeros -= 1
                    left += 1
                    

            res = max(res, (right-left))   
        
        return res

        

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        
        int n = nums.size();
        int left = 0;
        int zeroCount = 0;
        int res = 0;

        for (int right = 0; right < n; right++)
        {

            //Check 0's and 1's
            if (nums[right] == 0)
            {
                zeroCount++;
            }

            //Bounds check and checking 0's
            while (left < nums.size() && zeroCount > 1)
            {
                if (nums[left] == 0)
                {
                    zeroCount--;
                }

                left += 1;
            }

            //Check our max
            if (right - left + 1 > res)
            {
                res = right - left + 1;
            }
        }

        return res - 1;
    }
};